src/main/scala/org/dia/algorithms/mcc/MCCOps.scala

```diff
@@ -216,31 +248,42 @@ object MCCOps {
       edgeList: RDD[(Int, Iterable[MCCEdge])], iteration: Int,
       buckerSize: Int,
       minGraphLength: Int,
-      sc: SparkContext): Array[(Int, Iterable[MCCEdge])] = {
+      sc: SparkContext): Array[(Int, (Iterable[MCCEdge], mutable.MutableList[mutable.HashSet[String]]))] = {
     var iter = iteration
-    def startProcessing(obj: RDD[(Int, Iterable[MCCEdge])]): RDD[(Int, Iterable[MCCEdge])] = {
+    def startProcessing(obj: RDD[(Int, (Iterable[MCCEdge]))], iter: Int):
+    RDD[(Int, (Iterable[MCCEdge], mutable.MutableList[mutable.HashSet[String]]))] = {
       obj.map(x => processEdgePartition(x, iter, minGraphLength, buckerSize))
-        .filter(x => x._1 != -1)
+        .filter(x => {
+          val fw = new FileWriter("subgraphs.txt", true)
+          for (edge <- x._2._2) {
+            fw.write(edge.toString() + "\n")
+          }
+          fw.close()
+          x._1 != -1
+        })
         .reduceByKey((x, y) => {
```

Comment:
You can use case syntax in a lambda expression to unbox each tuple element into a separate variable.
e.g. 
.reduceByKey({ case ((xkey, xsubgraph), (ykey, ysubgraph))